src/PointwiseFunctions/Xcts/CenterOfMass.cpp

```diff
@@ -27,20 +29,22 @@ void center_of_mass_volume_integrand(
     gsl::not_null<tnsr::I<DataVector, 3>*> result,
     const Scalar<DataVector>& conformal_factor,
     const tnsr::i<DataVector, 3, Frame::Inertial>& deriv_conformal_factor,
-    const tnsr::I<DataVector, 3>& unit_normal) {
-  tenex::evaluate<ti::I>(result, 3. / (8. * M_PI) * 4. *
-                                     pow<3>(conformal_factor()) *
-                                     deriv_conformal_factor(ti::j) *
-                                     unit_normal(ti::I) * unit_normal(ti::J));
+    const tnsr::I<DataVector, 3>& coords) {
+  tenex::evaluate<ti::I>(
+      result,
+      3. / (4. * M_PI * pow<2>(magnitude(coords)())) *
```

Comment:
Move `magnitude(coords)` out of this expression into a separate line above. It allocates a DataVector, so it doesn't help to have it as part of this tensor expression.